pkg/api/types.go

```diff
@@ -466,6 +470,30 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// RBDVolumeSource represents a Rados Block Device Mount that lasts the lifetime of a pod
+type RBDVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that represents a collection of Ceph monitors
+	EndpointsName string `json:"endpoints"`
+	// Required: RBDImage is the rados image name
+	RBDImage string `json:"image"`
+	// Required: Filesystem type to mount.
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs"
+	// TODO: how do we prevent errors in the filesystem from compromising the machine
+	FSType string `json:"fsType,omitempty"`
+	// Optional: RBDPool is the rados pool name,default is rbd
+	RBDPool string `json:"pool"`
+	// Optional: RadosUser is the rados user name, default is admin
+	RadosUser string `json:"user"`
+	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
+	Keyring string `json:"keyring"`
+	// Optional: Secret is the authentication secret for RBDUser, default is empty.
+	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
```

Comment:
> On Apr 15, 2015, at 7:01 AM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > +	EndpointsName string `json:"endpoints"`
> > +	// Required: RBDImage is the rados image name
> > +	RBDImage string `json:"image"`
> > +	// Required: Filesystem type to mount.
> > +	// Must be a filesystem type supported by the host operating system.
> > +	// Ex. "ext4", "xfs", "ntfs"
> > +	// TODO: how do we prevent errors in the filesystem from compromising the machine
> > +	FSType string `json:"fsType,omitempty"`
> > +	// Optional: RBDPool is the rados pool name,default is rbd
> > +	RBDPool string `json:"pool"`
> > +	// Optional: RadosUser is the rados user name, default is admin
> > +	RadosUser string `json:"user"`
> > +	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
> > +	Keyring string `json:"keyring"`
> > +	// Optional: Secret is the authentication secret for RBDUser, default is empty.
> > +	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
> Vish asked a good question today:
> 
> If this volume type (and others) are expected to be used as PersistentVolumeSources, what does it mean to have an EndpointsName or a SecretName? PersistentVolumeSource is not namespaced, Endpoints and Secret are.
> 
> Does it mean that EndpointsName and SecretName will be resolved relative to the eventual PVClaim? That feels very odd - I have to put a secret in just the right place in my own namespace before using making a claim, but I don't know what type of PV I will actually be getting.
> 
I think the actor resolving the claim is the one responsible for ensuring the endpoints / secrets are available in the namespace with the claim.
> Or does it mean that EndpointsName and SecretName really need to be object refs and will be resolved in their own Namespaces (as per the object ref) without the claimant ever knowing of them? This feels cleaner to me. But we'd need to make sure that (non-Persistent) VolumeSource users of this same struct DON'T do cross-namespace stuff. Seems reasonable.
> 
It's possible, but vastly complicates the security story.  Suddenly any Node can access any secret in the entire cluster?  Solvable, just scares me.   
> Am I making sense?
> 
> —
> Reply to this email directly or view it on GitHub.
>